OpenSCAP/scap-security-guide
# [Enhancement] [RHEL/7] Enable OVAL 5.11 language support for RHEL-7 content if underlying system's "oscap" tool already supports OVAL 5.11

This patchset enables the support for OVAL language 5.11 constructs if the underlying system's "oscap" tool version supports OVAL 5.11 language version. If not, the (enable / disable) service OVAL checks are not included into the produced OVAL document and OVAL version is set to be the 5.10 one.

This change is reached via:
* https://github.com/OpenSCAP/scap-security-guide/commit/c4b4fdceee981ef09201c9ddfe83c686015de5d4 introduces a new ```RHEL/7/input/checks/oval_5.11``` subdirectory intended to hold OVAL checks that are utilizing some of the OVAL-5.11 language features (either ```systemdunitproperty / systemdunitdependency``` constructs or ```glob_to_regex``` function call). The ```RHEL/7/Makefile``` has been altered to check if underlying "oscap" tool version on the system supports OVAL 5.11 version, and if so, the checks from ```RHEL/7/input/checks/oval_5.11``` subdirectory are included into the final list of OVAL checks for the product. If the underlying "oscap" tool on the system doesn't support OVAL-5.11 language version yet (IOW it supports just OVAL-5.10 / OVAL-5.10.1), those service checks won't be included into the resulting / produced OVAL document for RHEL-7 product.

* https://github.com/OpenSCAP/scap-security-guide/commit/fd91798843c83d9114ed572e182ccf3ba2ee528f this patch applies the former enhancement to implement ```service_auditd_enabled``` and ```service_chronyd_enabled``` OVAL checks for RHEL-7's PCI-DSS profile

* https://github.com/OpenSCAP/scap-security-guide/commit/606771ee17813e2980bebbc23e3209aa66ce8c31 this patch applies the enhancement to include ```service_rsyslog_enabled``` and ```service_rdisc_disabled``` OVAL checks for RHEL-7's Common profile

* https://github.com/OpenSCAP/scap-security-guide/commit/ce424a4eda2a54b03d17b18018cb185d5ad89d83 this patch utilizes the enhancement to add ```service_oddjobd_disabled``` and ```service_qpidd_disabled``` OVAL checks for RHEL-7's Common profile

* https://github.com/OpenSCAP/scap-security-guide/commit/2cf6cbfb3efb3686c7afa2fcbd93e1e0b48bfd64 this patch uses the first patch to add ```service_autofs_disabled``` and ```service_ntpdate_disabled``` OVAL checks for RHEL-7's Common profile, and finally

* https://github.com/OpenSCAP/scap-security-guide/commit/f94924b26fe96ef04b048195d47ee49a96597ef1 uses the first change / patch to add ```service_atd_disabled``` and ```service_abrtd_disabled``` OVAL checks for RHEL-7's Common profile

Testing report:
------------------
The proposed change has been tested on RHEL-6, RHEL-7 (both with oscap not suppporting OVAL-5.11, and oscap supporting OVAL-5.11 language version), and on Fedora 20 (with oscap suppporting OVAL-5.11 language version), and based on the testing it is working fine (```make validate``` still passes on RHEL-6 and Fedora products. When building on RHEL-7 without 5.11 support, the ```service enabled / disabled``` checks aren't included into the resulting OVAL document, when built on RHEL-7 with 5.11 support ```service enabled / disabled``` checks are included into the resulting OVAL document).

Also, each of the ```service * enabled / disabled``` changes have been tested && recognized as working properly on RHEL-7 system with oscap supporting 5.11 OVAL language version.

Note:
-------
This patches *didn't intentionally* include the ```testcheck.py``` utility into the newly introduced ```RHEL/7/input/checks/oval_5.11``` subdirectory - since this would require the ```testcheck.py``` utility to be enhanced to support two possible OVAL language versions (OVAL-5.10, and OVAL-5.11), depending on the version of the "oscap" tool, which was used to build RHEL-7 content. Since this is non-trivial change and would require to introduce yet another repository changes, I have decided to keep / postpone this enhancement fo some of the future PRs. Therefore, to check if the included RHEL-7 ```service enabled / disabled``` checks are truly working the reviewer will need to:
* build RHEL-7 content with "oscap" tool supporting OVAL-5.11 language version (e.g. "oscap" for RHEL-7 from the Copr repository),
* run the service checks manually either via "oscap" tool directly or by utilizing the "scap-workbench" GUI tool (this is what was used to test the changes).

Please review.

Thank you, Jan.
